jobs/tests/templates/tests/etcd

```diff
@@ -0,0 +1,278 @@
+    fi
+
+    if [[ $(etcdctl --user root:tests --endpoints=$url get pokemon --print-value-only) != "psyduck" ]]; then
+        echo "!! pokemon key was not initialized properly"
+        rc=1
+    else
+        echo "Got correct value for key pokemon"
+    fi
+
+    if [[ $(etcdctl --user root:tests --endpoints=$url get avengers --print-value-only) != "endgame" ]]; then
+        echo "!! avengers key was not initialized properly"
+        rc=1
+    else
+        echo "Got correct value for key avengers"
+    fi
+
+    if [[ $(etcdctl --user root:tests --endpoints=$url get extraKey --print-value-only) != "" ]]; then
+        etcdctl --user root:tests --endpoints=$url get extraKey --print-value-only
+        echo "!! Overwrite did not work correctly"
+        rc=1
+    else
+        echo "Overwrite works. extraKey doesn't exist"
+    fi
+
+    echo
+    return $rc
+}
+
+#Getting values from keys for etcd-tls
+verifyValsTls() {
+    local url=$1
+    local rc=0
+    if [[ $(etcdctl --cacert=/var/vcap/jobs/tests/data/etcd-tls/tls/cert.pem --key=/var/vcap/data/targets/etcd/key.pem --cert=/var/vcap/data/targets/etcd/cert.pem --endpoints=$url get foo --print-value-only) != "bar" ]]; then
+        echo "!! foo key was not initialized properly"
+        rc=1
+    else
+        echo "Got correct value for key foo"
+    fi
+
+    if [[ $(etcdctl --cacert=/var/vcap/jobs/tests/data/etcd-tls/tls/cert.pem --key=/var/vcap/data/targets/etcd/key.pem --cert=/var/vcap/data/targets/etcd/cert.pem --endpoints=$url get pokemon --print-value-only) != "psyduck" ]]; then
+        echo "!! pokemon key was not initialized properly"
+        rc=1
+    else
+        echo "Got correct value for key pokemon"
+    fi
+
+    if [[ $(etcdctl --cacert=/var/vcap/jobs/tests/data/etcd-tls/tls/cert.pem --key=/var/vcap/data/targets/etcd/key.pem --cert=/var/vcap/data/targets/etcd/cert.pem --endpoints=$url get avengers --print-value-only) != "endgame" ]]; then
+        echo "!! avengers key was not initialized properly"
+        rc=1
+    else
+        echo "Got correct value for key avengers"
+    fi
+
+    if [[ $(etcdctl --cacert=/var/vcap/jobs/tests/data/etcd-tls/tls/cert.pem --key=/var/vcap/data/targets/etcd/key.pem --cert=/var/vcap/data/targets/etcd/cert.pem --endpoints=$url get extraKey --print-value-only) != "" ]]; then
+        echo "!! Overwrite did not work correctly"
+        rc=1
+    else
+        echo "Overwrite works. extraKey doesn't exist"
+    fi
+
+    echo
+    return $rc
+}
+
+#Prints all keys and values
+dumpKeys() {
+    local url=$1
+    echo "Printing all keys and values"
+    etcdctl --endpoints=$url --user root:tests get "" --prefix=true
+    echo
+}
+
+#Prints all keys and values for etcd-tls
+dumpKeysTls() {
+    local url=$1
+    echo "Printing all keys and values"
+    etcdctl --cacert=/var/vcap/jobs/tests/data/etcd-tls/tls/cert.pem --key=/var/vcap/data/targets/etcd/key.pem --cert=/var/vcap/data/targets/etcd/cert.pem --endpoints=$url get "" --prefix=true
```

Comment:
Investigate `ETCD_OPTIONS` as an environment variable that you set for the TLS test suite only.  You should be able to drop half of these functions and just use one common set of primitives.